src/main/java/com/amazon/opendistroforelasticsearch/security/OpenDistroSecurityPlugin.java

```diff
@@ -706,7 +697,8 @@ public AsyncSender interceptSender(AsyncSender sender) {
             Environment environment, NodeEnvironment nodeEnvironment, NamedWriteableRegistry namedWriteableRegistry,
             IndexNameExpressionResolver indexNameExpressionResolver, Supplier<RepositoriesService> repositoriesServiceSupplier) {
 
-        if(sslOnly) {
+        openDistroSSLConfig.registerClusterSettingsChangeListener(clusterService.getClusterSettings());
```

Comment:
Registering the setting even in not SSL modes since the cluster setting can be modified using the PUT _cluster/setting api. 
However, we already have a check inside the ```OpenDistroSSLConfig``` class to check if ssl_only mode is enabled. ```isDualModeEnabled()``` function returns true only when ssl_only is enabled and dual_mode is enabled